frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx

```diff
@@ -177,6 +177,18 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
     (isEditing && combinedDisk.getVolumeMode()) || null,
   );
 
+  React.useEffect(() => {
+    if (source.requiresPVC()) {
+      const pvcNameDataVolume = dataVolume.getPersistentVolumeClaimName();
+      const pvcNamespaceDataVolume = dataVolume.getPersistentVolumeClaimNamespace();
+      const pvc = persistentVolumeClaims?.data.find(
+        ({ metadata }) =>
+          metadata?.name === pvcNameDataVolume && metadata?.namespace === pvcNamespaceDataVolume,
+      );
+      setVolumeMode((value) => VolumeMode.fromString(pvc?.spec?.volumeMode) || value);
+    }
+  }, [dataVolume, persistentVolumeClaims, source]);
```

Comment:
please don't use dataVolume and persistentVolumeClaims as dependencies for useEffect as they can have a different reference each rerender and will fire every time. Please guard it with loaded status of persistentVolumeClaims, source, and volumeMode

Reply:
It's been added automatically by the linter, basically, there aren't any side effects that can cause performance issues. I agree that if there is a performance issue we should avoid it, but I don't think this is the case.

Reply:
the linter does't check for these things, only in very limited scope. It is not like you can measure the performance issues, but the code isn't crisp and fires setVolumeMode needlesly.

Reply:
volumeMode is a string, a primitive type, useState can compare it easily,  it won't get updated if the value is the same, so in this case, I think no need for concern :)